app/assets/src/components/common/DetailsSidebar/BulkDownloadDetailsMode/AdvancedDownloadTab.jsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import copy from "copy-to-clipboard";
+import cx from "classnames";
+
+import PropTypes from "~/components/utils/propTypes";
+import CopyIcon from "~ui/icons/CopyIcon";
+import BasicPopup from "~/components/BasicPopup";
+import { withAnalytics } from "~/api/analytics";
+
+import cs from "./bulk_download_details_mode.scss";
+
+export default class AdvancedDownloadTab extends React.Component {
+  state = {
+    tooltip: "",
+  };
+
+  hasCloudCommand = () => {
+    const { bulkDownload } = this.props;
+    return (
+      bulkDownload.status === "success" &&
+      bulkDownload.presigned_output_url !== null
+    );
+  };
+
+  getCloudCommand = () => {
+    const { bulkDownload } = this.props;
+
+    if (
+      bulkDownload.status === "waiting" ||
+      bulkDownload.status === "running"
+    ) {
+      return "Bulk download is not yet complete.";
+    }
+
+    if (bulkDownload.status === "error") {
+      return "Bulk download failed. Please contact us for help.";
+    }
+
+    if (
+      bulkDownload.status === "success" &&
+      bulkDownload.presigned_output_url === null
+    ) {
+      return "Failed to generate command. Please contact us for help.";
+    }
+
+    return `curl -L "${bulkDownload.presigned_output_url}" > "${
+      bulkDownload.download_name
+    }.tar.gz"\
+        && mkdir -p "${bulkDownload.download_name}"\
+        && tar -zvxf "${bulkDownload.download_name}.tar.gz" -C "${
+      bulkDownload.download_name
+    }"\
+      `;
+  };
+
+  copyCommandToClipboard = () => {
+    copy(this.getCloudCommand());
+    this.setState({
+      tooltip: "Copied command to clipboard",
+    });
+  };
+
+  resetTooltip = () => {
+    this.setState({
+      tooltip: "",
+    });
+  };
+
+  render() {
+    const { bulkDownload } = this.props;
+
+    const handleCloudCommandClick = this.hasCloudCommand()
+      ? withAnalytics(
+          this.copyCommandToClipboard,
+          "AdvancedDownloadTab_copy-cloud-command-link_clicked",
+          {
+            bulkDownloadId: bulkDownload.id,
+            fileSize: bulkDownload.file_size,
+          }
+        )
+      : null;
+
+    const cloudCommandContainer = (
+      <div
+        className={cx(
+          cs.cloudCommandContainer,
+          this.hasCloudCommand() && cs.enabled
+        )}
+        onClick={handleCloudCommandClick}
+        onMouseLeave={this.resetTooltip}
+      >
+        <div className={cs.command}>{this.getCloudCommand()}</div>
+        <CopyIcon className={cs.icon} />
+      </div>
+    );
+
+    return (
+      <div className={cs.advancedDownloadTab}>
+        <div className={cs.header}>Download to your cloud machine:</div>
```

Comment:
Agree, suggest just saying "your machine"?

Reply:
Good point! Although "your machine" seems confusing because downloading the file normally is also to your machine. I'll change it to "Download via command line" if that sounds okay.

Reply:
sounds good to me